src/java/arjdbc/jdbc/RubyJdbcConnection.java

```diff
@@ -2016,132 +2019,84 @@ protected void setStatementParameters(final ThreadContext context,
         final Connection connection, final PreparedStatement statement,
         final List<?> binds) throws SQLException {
 
-        final Ruby runtime = context.getRuntime();
-
         for ( int i = 0; i < binds.size(); i++ ) {
-            // [ [ column1, param1 ], [ column2, param2 ], ... ]
-            Object param = binds.get(i); IRubyObject column = null;
-            if ( param instanceof RubyArray ) {
-                final RubyArray _param = (RubyArray) param;
-                column = _param.eltInternal(0); param = _param.eltInternal(1);
-            }
-            else if ( param instanceof List ) {
-                final List<?> _param = (List<?>) param;
-                column = (IRubyObject) _param.get(0); param = _param.get(1);
-            }
-            else if ( param instanceof Object[] ) {
-                final Object[] _param = (Object[]) param;
-                column = (IRubyObject) _param[0]; param = _param[1];
-            }
-
-            setStatementParameter(context, runtime, connection, statement, i + 1, param, column);
+            setStatementParameter(context, connection, statement, i + 1, (IRubyObject) binds.get(i));
```

Comment:
if binds is an `RubyArray` it will attempt auto toJava conversion - which we might not want
... maybe we can just assume binds is always a RubyArray, right?

Reply:
Yeah, as far as I can tell `binds` should always be a `RubyArray`. If we assume that, what changes should I make here?

Reply:
would start with the RubyArray path that got deleted - just cast directly, for now and let the API cleanup happen later
~ `((RubyArray) binds).eltIternal(i)` ... so its clear what were passing here, guess we never passed anything else